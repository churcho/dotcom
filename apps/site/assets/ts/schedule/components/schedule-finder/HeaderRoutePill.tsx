import React, { ReactElement } from "react";
import isSilverLine, {
  isSilverLineWaterfront
} from "../../../helpers/silver-line";
import { RouteType } from "../../../__v3api";

interface Props {
  id: string;
  type: RouteType;
  name: string;
}

const HeaderRoutePill = ({
  id,
  type,
  name
}: Props): ReactElement<HTMLElement> | null =>
  type === 3 ? (
    <div className="m-route-pills">
      <div
        className={`h1 schedule-finder__modal-route-pill u-bg--${
          isSilverLine(id) ? "silver-line" : "bus"
        }`}
      >
        {isSilverLineWaterfront(id) ? "SL" : name}
      </div>
    </div>
  ) : null;

export default HeaderRoutePill;
