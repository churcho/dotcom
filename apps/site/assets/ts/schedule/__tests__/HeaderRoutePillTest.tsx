import React from "react";
import renderer, { act } from "react-test-renderer";
import HeaderRoutePill from "../components/schedule-finder/HeaderRoutePill";

describe("HeaderRoutePill", () => {
  it("renders for bus", () => {
    let tree;
    act(() => {
      tree = renderer.create(<HeaderRoutePill id="123" name="123" type={3} />);
    });
    expect(tree).toMatchSnapshot();
  });

  it("renders SL routes appropriately", () => {
    let tree;
    act(() => {
      tree = renderer.create(<HeaderRoutePill id="741" name="SL1" type={3} />);
    });
    expect(tree).toMatchSnapshot();
  });

  it("renders SL Waterfront appropriately", () => {
    let tree;
    act(() => {
      tree = renderer.create(
        <HeaderRoutePill
          id="746"
          name="Silver Line Way - South Station"
          type={3}
        />
      );
    });
    expect(tree).toMatchSnapshot();
  });

  it("returns null for non-bus service", () => {
    const tree = HeaderRoutePill({ id: "Red", type: 1, name: "RL" });
    expect(tree).toBeNull();
  });
});
