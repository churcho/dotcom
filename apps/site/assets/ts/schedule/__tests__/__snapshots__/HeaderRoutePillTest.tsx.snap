// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderRoutePill renders SL routes appropriately 1`] = `
<div
  className="m-route-pills"
>
  <div
    className="h1 schedule-finder__modal-route-pill u-bg--silver-line"
  >
    SL1
  </div>
</div>
`;

exports[`HeaderRoutePill renders for bus 1`] = `
<div
  className="m-route-pills"
>
  <div
    className="h1 schedule-finder__modal-route-pill u-bg--bus"
  >
    123
  </div>
</div>
`;
